src/persist/src/runtime.rs

```diff
@@ -88,34 +88,24 @@ where
     let (tx, rx) = crossbeam_channel::unbounded();
     let metrics = Arc::new(Metrics::register_with(reg));
 
-    // Any usage of S3Blob requires a runtime context to be set. `Indexed::new`
-    // use the blob impl to start the recovery process, so make sure this stays
-    // early.
-    let pool = match pool {
+    let async_runtime = match async_runtime {
         Some(pool) => pool,
-        None => Arc::new(Runtime::new()?),
+        None => Arc::new(AsyncRuntime::new()?),
     };
-    let pool_guard = pool.enter();
 
     // Start up the runtime.
-    let blob = BlobCache::new(build, metrics.clone(), blob);
-    let maintainer = Maintainer::new(blob.clone(), pool.clone());
+    let blob = BlobCache::new(build, metrics.clone(), async_runtime.clone(), blob);
+    let maintainer = Maintainer::new(blob.clone(), async_runtime.clone());
     let indexed = Indexed::new(log, blob, maintainer, metrics.clone())?;
     let mut runtime = RuntimeImpl::new(config.clone(), indexed, rx, metrics.clone());
     let id = RuntimeId::new();
-    let runtime_pool = pool.clone();
     let impl_handle = thread::Builder::new()
         .name("persist:runtime".into())
-        .spawn(move || {
```

Comment:
We don't need to "enter" the runtime here anymore because the `BlobCache` (and maybe all other places that need a runtime) now make sure they have one themselves?